File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        auto n = nums.size();
        auto rotated = false;

        for(auto i=1; i<n; i++){
            if(nums[i] >= nums[i-1]) continue;
            if(rotated) return false;
            rotated = true;
        }

        return (!rotated || nums[0] >= nums[n-1]);
    }
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n, rotated = len(nums), False
        for i in range(1, n):
            if nums[i] >= nums[i-1]: continue
            if rotated: return False
            rotated = True

        return not rotated or nums[0] >= nums[n-1]